ui/src/app/pages/signed-in/signed-in.tsx

```diff
@@ -0,0 +1,171 @@
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', checkOpsBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', checkOpsBeforeUnload);
+    };
+  }, []);
+
+  useEffect(() => {
+    const subs = [];
+    // This handles detection of Angular-based routing data.
+    subs.push(routeConfigDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    // This handles detection of React-based routing data. During migrations,
+    // we assume React routing data will be set deeper/later in the component
+    // hierarchy, therefore it will generally take precedence over React.
+    subs.push(routeDataStore.subscribe(({minimizeChrome}) => {
+      setHideFooter(minimizeChrome);
+    }));
+    setSubscriptions(subs);
+
+    return () => {
+      /*
+       * TODO RW-1920: we will only need one subscription after routeConfigDataStore and
+       *  routeDataStore are collapsed, and then we won't have to duck-type this
+       */
+      for (const s of subscriptions) {
+        s.unsubscribe();
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    /*
+     * TODO RW-6726 We want to block app rendering on the presence of server config data
+     *  so that route components don't try to lookup config data before it's available.
+     *  We should holistically figure out a Reacty way to do this.
+     *  See discussion on https://github.com/all-of-us/workbench/pull/4713
+     */
+    const checkStoresLoaded = async() => {
+      if (serverConfig.config) {
+        setServerConfigInitialized(true);
+        if (!profileStore.get().profile) {
+          await profileStore.get().load();
+        }
+        setInstitutionCategoryState(profileStore.get().profile.verifiedInstitutionalAffiliation);
+        if (hasRegisteredAccess(profileStore.get().profile.accessTierShortNames)) {
+          if (!cdrVersions) {
+            const cdrVersionsByTier = await cdrVersionsApi().getCdrVersionsByTier();
+            cdrVersionStore.set(cdrVersionsByTier);
+            setCdrVersionsInitialized(true);
+          } else {
+            setCdrVersionsInitialized(true);
+          }
+        } else {
+          setCdrVersionsInitialized(true);
+        }
+      }
+    };
+
+    checkStoresLoaded();
+  }, []);
+
+  const signOut = (continuePath?: string): void => {
+    window.localStorage.setItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE, null);
+    props.onSignOut();
+    props.signOut();
+    navigateSignOut(continuePath);
+  };
+
+  return <FlexColumn
+    style={{
+      minHeight: '100vh',
+      /* minimum supported width is 1300, this allows 20px for the scrollbar */
+      minWidth: '1280px'
+    }}
+    data-test-id='signed-in'
+  >
+    <NavBar/>
+    <FlexRow style={{position: 'relative', flex: '1 0 auto'}}>
+      <div
+          style={styles.backgroundImage}
+      />
+      {cdrVersionsInitialized && serverConfigInitialized &&
```

Comment:
super nit: I would probably reverse the order of these two booleans to better match the load / dependency ordering